### PR TITLE
Allow to launch multiple apps via try now page

### PR DESCRIPTION
**Pull request by jfaltermeier**
Adds a drop down to select different applications to launch. 
ksnip_20221017-150315

ksnip_20221017-150316

The default selection may be influenced via a url param, e.g. `https://theia.cloud.192.168.39.19.nip.io/?appDef=coffee-editor`

**For testing the try now page itself without Theia.cloud service+operator**

Open `node/try-now-page/public/config.js` and uncomment the `additionalApps`
Build and launch the webpage with `npm run build && npm run start -w try-now-page`

`http://localhost:3000` should have the default Theia Blueprint selected. 

`http://localhost:3000/?appDef=coffee-editor` should have coffee-editor preselected. 

When attempting to launch an app, you have to check the console out to verify the app, which is attempted to be launched:
```
Launching theia-cloud-demo
Launching coffee-editor
Launching cdt-cloud-demo
```

**For testing via minikube**

Build the docker images (https://github.com/eclipsesource/theia-cloud/blob/main/doc/docs/platforms/Minikube.md#testing-local-images) and adjust the images in `helm/theia.cloud/valuesMinikube.yaml` accordingly (remember to also switch from `theiacloud/theia-cloud-landing-page` to `theiacloud/theia-cloud-try-now-page`)
Besides that also disable keycloak:
```
keycloak:
  enable: false
```
and uncomment the `additionalApps` and set `ephemeralStorage` to `true`:
```
  ephemeralStorage: true
  additionalApps: 
    coffee-editor:
      label: "Coffee Editor"
    cdt-cloud-demo:
      label: "CDT.cloud Blueprint"
```
Install Theia.cloud via helm with the `valuesMinikube.yaml`

Then adjust the host in `demo/k8s/appdefinitions/cdt.yaml` and `demo/k8s/appdefinitions/coffee-editor.yaml` to match your minikube instances host. 
Install both yamls with `kubectl apply -f cdt.yaml && kubectl apply -f coffee-editor.yaml`

Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: node/try-now-page/src/App.tsx [hunk 1]
```diff
@@ -19,9 +24,39 @@ function App(): JSX.Element {
...
+      initialised = true;
+      const element = document.getElementById('selectapp');
+      const urlParams = new URLSearchParams(window.location.search);
+      // eslint-disable-next-line no-null/no-null
+      if (element !== null && urlParams.has('appDef')) {
+        const defaultSelection = urlParams.get('appDef');
+        // eslint-disable-next-line no-null/no-null
+        if (defaultSelection !== null) {
+          (element as HTMLSelectElement).value = defaultSelection;
+          setSelectedAppName((element as HTMLSelectElement).options[(element as HTMLSelectElement).selectedIndex].text);
```
Comment (AlexandraBuzila) [inline node/try-now-page/src/App.tsx:46]: If the `defaultSelection` is a value not present in `config.additionalApps`, the `selectedIndex` will be -1 and this line will throw an error. 
In that case, the default app definition should probably be used.

**Review by AlexandraBuzila — CHANGES_REQUESTED** [review]
Looking good! There's one corner case that should still be handled, to avoid breaking the page when the `appDef` value provided via url is invalid.

File: node/try-now-page/src/App.tsx (hunk 1 above)
Comment (jfaltermeier) [reply node/try-now-page/src/App.tsx:46]: Thanks. I've pushed an additional commit that will check whether the default selection is valid.

**Comment by jfaltermeier** [comment]
@AlexandraBuzila When would you have time for a short re-review?
We will do a milestone release when this one is merged.

**Review by AlexandraBuzila — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**